sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/csv/CSVExprUtils.scala

```diff
@@ -79,4 +81,39 @@ object CSVExprUtils {
         throw new IllegalArgumentException(s"Delimiter cannot be more than one character: $str")
     }
   }
+
+  /**
+   * Helper method that converts string representation of a character sequence to actual
+   * delimiter characters. The input is processed in "chunks", and each chunk is converted
+   * by calling [[CSVExprUtils.toChar()]].  A chunk is either:
+   * <ul>
+   *   <li>a backslash followed by another character</li>
+   *   <li>a non-backslash character by itself</li>
+   * </ul>
+   * , in that order of precedence. The result of the converting all chunks is returned as
+   * a [[String]]
+   *
+   * @param str the string representing the sequence of separator characters
+   * @return a [[String]] representing the multi-character delimiter
+   * @throws IllegalArgumentException if any of the individual input chunks are illegal
+   */
+  def toDelimiterStr(str: String): String = {
+    import scala.collection.mutable.StringBuilder
```

Comment:
You can import this at the top for consistency but honestly it's probably overkill to use a stringbuilder. This is rarely more than a few characters and is done once.

Reply:
Switched to a `ListBuffer` instead, but I'm open to other suggestions.  I couldn't figure out a nice, functional way to do this block.

Reply:
```
var delimiter = ""
...
  delimiter += toChar(...)
``
? that kind of thing is even simpler, and given just a few small concatenations, doesn't matter for perf.

Reply:
Done.